src/mono/mono/mini/interp/transform.c

```diff
@@ -3404,6 +3404,93 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 		return FALSE;
 	}
 
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+	/*
+	* We need to modify the signature of the swiftcall calli to account for the lowering of Swift structs.
+	* This is done by replacing struct arguments on stack with a lowered sequence and updating the signature.
+	*/
+	if (csignature->pinvoke && mono_method_signature_has_ext_callconv (csignature, MONO_EXT_CALLCONV_SWIFTCALL)) {
+		g_assert (!csignature->hasthis); // P/Invoke calls shouldn't contain 'this'
+
+		// Save the function pointer
+		StackInfo sp_fp = td->sp [-1];
+		--td->sp;
+
+		// Save the old arguments				
+		td->sp -= csignature->param_count;
+		StackInfo *sp_old_params = (StackInfo*) mono_mempool_alloc (td->mempool, sizeof (StackInfo) * csignature->param_count);
+		memcpy (sp_old_params, td->sp, sizeof (StackInfo) * csignature->param_count);
+
+		GArray *new_params = g_array_new (FALSE, FALSE, sizeof (MonoType*));
```

Comment:
Can you use `g_array_sized_new` with an initial size equal to the old param count (plus 4?).  that would avoid resizing most of the time.

Reply:
Yes, I changed it to use the old param count since we are unsure ahead-of-time if any lowering will occur.